src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1412,50 +1411,19 @@ private void RestoreOperatingEnvironment()
 
                 // Restore the saved environment variables.
                 SetEnvironmentVariableBlock(_requestEntry.RequestConfiguration.SavedEnvironmentVariables);
-                NativeMethodsShared.SetCurrentDirectory(_requestEntry.RequestConfiguration.SavedCurrentDirectory);
+                _requestEntry.TaskEnvironment.ProjectDirectory = new AbsolutePath(_requestEntry.RequestConfiguration.SavedCurrentDirectory, ignoreRootedCheck: true);
             }
         }
 
         /// <summary>
         /// Sets the environment block to the set of saved variables.
         /// </summary>
-        private void SetEnvironmentVariableBlock(FrozenDictionary<string, string> savedEnvironment)
+        private void SetEnvironmentVariableBlock(IReadOnlyDictionary<string, string> savedEnvironment)
         {
-            FrozenDictionary<string, string> currentEnvironment = CommunicationsUtilities.GetEnvironmentVariables();
-            ClearVariablesNotInEnvironment(savedEnvironment, currentEnvironment);
-            UpdateEnvironmentVariables(savedEnvironment, currentEnvironment);
-        }
-
-        /// <summary>
-        /// Clears from the current environment any variables which do not exist in the saved environment
-        /// </summary>
-        private void ClearVariablesNotInEnvironment(FrozenDictionary<string, string> savedEnvironment, FrozenDictionary<string, string> currentEnvironment)
-        {
-            foreach (KeyValuePair<string, string> entry in currentEnvironment)
-            {
-                if (!savedEnvironment.ContainsKey(entry.Key))
-                {
-                    Environment.SetEnvironmentVariable(entry.Key, null);
-                }
-            }
-        }
-
-        /// <summary>
-        /// Updates the current environment with values in the saved environment which differ or are not yet set.
-        /// </summary>
-        private void UpdateEnvironmentVariables(FrozenDictionary<string, string> savedEnvironment, FrozenDictionary<string, string> currentEnvironment)
-        {
-            foreach (KeyValuePair<string, string> entry in savedEnvironment)
-            {
-                // If the environment doesn't have the variable set, or if its value differs from what we have saved, set it
-                // to the saved value.  Doing the comparison before setting is faster than unconditionally setting it using
-                // the API.
-                string value;
-                if (!currentEnvironment.TryGetValue(entry.Key, out value) || !String.Equals(entry.Value, value, StringComparison.Ordinal))
```

Comment:
this is refactored in through driver -> communicationsutilities

new: the comparison depends on env var case sensitivity on the OS